src/stores/installed-apps-api.ts

```diff
@@ -0,0 +1,39 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+
+export type InstalledAppsState = InstalledApps & InstalledTerminals;
+
+export const installedAppsInitialState: InstalledAppsState = {
+	vscode: false,
+	phpstorm: false,
+	webstorm: false,
+	windsurf: false,
+	cursor: false,
+	terminal: false,
+	iterm: false,
+	ghostty: false,
+	warp: false,
+};
+
+export const installedAppsApi = createApi( {
+	reducerPath: 'installedAppsApi',
+	baseQuery: fetchBaseQuery(),
+	tagTypes: [ 'InstalledApps' ],
+	endpoints: ( builder ) => ( {
+		getInstalledApps: builder.query< InstalledAppsState, void >( {
+			queryFn: async () => {
+				try {
+					const installedApps = await getIpcApi().getInstalledApps();
+					return { data: installedApps as InstalledAppsState };
```

Comment:
```suggestion
					return { data: installedApps };
```
`getIpcApi().getInstalledApps()` seems to only return data about editors, not about terminals. We could either call `getIpcApi().getInstalledTerminals()`, too, and await both promises, or we refactor `ipc-handlers` to return data about editors and terminals from a single function.

Or, alternatively, we add a second query/endpoint to fetch data about installed terminals.

Reply:
Both `getInstalledApps()` and `getInstalledTerminals()` are using `isInstalled`, so I'd much prefer going the refactoring `ipc-handlers` route and moving everything to one place.